tpm.c

```diff
@@ -344,3 +345,31 @@ EFI_STATUS tpm_measure_variable(CHAR16 *VarName, EFI_GUID VendorGuid, UINTN VarS
 	return tpm_record_data_measurement(VarName, VendorGuid, VarSize,
 					   VarData);
 }
+
+EFI_STATUS
+fallback_should_prefer_reset(void)
+{
+	EFI_STATUS status;
+	efi_tpm_protocol_t *tpm;
+	efi_tpm2_protocol_t *tpm2;
+
+	status = LibLocateProtocol(&tpm2_guid, (VOID **)&tpm2);
+	if (status == EFI_SUCCESS) {
+		BOOLEAN old_caps;
+		EFI_TCG2_BOOT_SERVICE_CAPABILITY caps;
+		status = tpm2_get_caps(tpm2, &caps, &old_caps);
+		if (EFI_ERROR(status))
+			return status;
+
+		if (tpm2_present(&caps, old_caps))
+			return EFI_SUCCESS;
+	} else {
+		status = LibLocateProtocol(&tpm_guid, (VOID **)&tpm);
+		if (EFI_ERROR(status))
+			return status;
+
+		if (tpm_present(tpm))
+			return EFI_SUCCESS;
+	}
+	return EFI_NOT_FOUND;
+}
```

Comment:
This duplicates part of the logic in tpm_log_event_raw(), I wonder if we could as a follow-up consolidate this and have the shared part in a helper function or something.